src/EditProvider.ts

```diff
@@ -99,16 +100,84 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    path: string = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      /**
+       * Use a different `.unibeautify` config when vscode config is set and
+       * define if the project config should extend the vscode config.
+       *
+       * @argument {unibeautify.defaultConfig} string path to a file
+       * @argument {unibeautify.extendConfig} boolean extend or not
+       */
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
```

Comment:
Use the `UnibeautifyVSCodeSettings` interface from https://github.com/Unibeautify/vscode/blob/master/src/index.ts#L65-L67

I think you will need to move `UnibeautifyVSCodeSettings` from `index.ts` to this file then import into `index.ts`.

```diff
- const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+ const vscodeSettings: UnibeautifyVSCodeSettings = vscode.workspace.getConfiguration("unibeautify");
```

Reply:
How do you mean this, I did this now:

```
import { UnibeautifyVSCodeSettings } from "./index";
.......
      const vscodeSettings: UnibeautifyVSCodeSettings = vscode.workspace.getConfiguration(
        "unibeautify"
      );
      let defaultConfigOverride = vscodeSettings.defaultConfig;
```

ang got this tslint error:
```
[ts]
Der Typ "WorkspaceConfiguration" kann dem Typ "UnibeautifyVSCodeSettings" nicht zugewiesen werden.
  Die Eigenschaft "enabled" fehlt im Typ "WorkspaceConfiguration".
const vscodeSettings: UnibeautifyVSCodeSettings
```

Reply:
Ahh sorry. I always forget that. I should turn my computer language to english!

The problem here ist, that on the next line, it does not find the `.defaultConfig` prop. I changed it a little bit to get it working. Are you fine with that:

```typescript
      const vscodeSettings: UnibeautifyVSCodeSettings = <any>vscode.workspace.getConfiguration(
        "unibeautify"
      );
      const defaultConfigFile = (vscodeSettings as any).defaultConfig;
```